GitUI/CommandsDialogs/FormDeleteBranch.cs

```diff
@@ -48,15 +48,22 @@ protected override void OnRuntimeLoad(EventArgs e)
             base.OnRuntimeLoad(e);
 
             Branches.BranchesToSelect = Module.GetRefs(RefsFilter.Heads).ToList();
-            foreach (string branch in Module.GetMergedBranches())
+            if (AppSettings.DontConfirmDeleteUnmergedBranch)
             {
-                if (!branch.StartsWith("* "))
-                {
-                    _mergedBranches.Add(branch.Trim());
-                }
-                else if (!branch.StartsWith("* ") || !DetachedHeadParser.IsDetachedHead(branch[2..]))
+                _currentBranch = Module.GetSelectedBranch();
```

Comment:
This comment probably what I missed at the first look, should be cleared too
```suggestion
                // No check for merged branches is needed
                _currentBranch = Module.GetSelectedBranch();
                _mergedBranches.Clear();
```